downstream/modules/platform/con-controller-api-identifier-format-protocol.adoc

```diff
@@ -0,0 +1,38 @@
+[id="controller-api-identifier-format-protocol"]
+
+= Identifier format protocol
+
+Resources are identifiable by their unique keys, which are tuples of resource fields. 
+Every resource is guaranteed to have its primary key number alone as a unique key, but there might be many other unique keys. 
+A resource can generate an identifier format and, therefore, have a named URL if it has at least one unique key that satisfies the following rules:
+
+. The key must contain only fields that are either the `name` field, or text fields with a finite number of possible choices (such as credential type resource's `kind` field).
+. The only allowed exceptional fields that breaks the preceding rule is a many-to-one related field relating to a resource other than itself, which is also allowed to have a slug.
+
+If there are resources `Foo` and `Bar`, both `Foo` and `Bar` contain a name field and a choice field that can only have value "yes" or "no". 
```

Comment:
I still don't know why people continue using Foo and Bar, considering it's origin.